common/background_exception.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+// Standard C++ includes
+#include <stdexcept>
+
+namespace BoBRobotics {
+//----------------------------------------------------------------------------
+// BoBRobotics::BackgroundException
+//----------------------------------------------------------------------------
+//! A wrapper for passing exceptions between threads (i.e. a background thread to the main one)
+class BackgroundException {
```

Comment:
It was only because I also use this class in ``Video::NetSink`` which isn't a ``Threadable``. I guess also because it's a general thing that we might want to use for other threaded code in future.

Reply:
Ahh I didn't notice that - in which case this is the correct solution